src/debuggerCommunicationServer.ts

```diff
@@ -36,16 +55,22 @@ export class DebuggerCommunicationServer {
         this.simulatorWebview = webviewPanel;
     }
 
-    // Emit Buttons Inputs Events
-    public emitButtonPress(newState: string): void {
-        console.log(`Emit Button Press: ${newState} \n`);
-        this.serverIo.emit("button_press", newState);
-    }
-
-    // Emit Sensors Inputs Events
-    public emitSensorChanged(newState: string): void {
-        console.log(`Emit Sensor Changed: ${newState} \n`);
-        this.serverIo.emit("sensor_changed", newState);
+    public emitInputChanged(newState: string): void {
+        if (this.isWaitingResponse) {
+            this.currentCall.push(() => {
+                this.serverIo.emit(
+                    DEBUGGER_MESSAGES.EMITTER.INPUT_CHANGED,
+                    newState
+                );
+                this.isWaitingResponse = true;
```

Comment:
Do you know if this runs single-threaded? (I _think_ it does?) If not, we might want to consider atomic operations when getting/setting the value of `isWaitingResponse`.